server/controllers/authController.js

```diff
@@ -168,36 +108,94 @@ const handleLogin = async (req, res) => {
   }
 };
 
-//@endpoint GET /user/profile
-//@desc authenticate user token and send back user info
-//@access PRIVATE
+// ----------------------------------------SIGNUP----------------------------------------
+//@endpoint POST /SignUp
+//@desc Sign up user info.
+//@access PUBLIC
 
-const handleGetUser = async (req, res, next) => {
-  //from middleware.
-  let id = req.user.id;
-  try {
-    const db = getConnection().db(dbName);
-    //see if you find the user.
-    let checkForUser = await db
-      .collection(collectionUsers)
-      .findOne({ _id: ObjectId(id) });
-    res.status(200).json({
-      status: 200,
-      message: "User remains signed in. Token verification accepted.",
-      username: checkForUser.username,
-      accessToken: checkForUser.accessToken,
-      _id: checkForUser._id,
-      profileImage: checkForUser.profileImage,
+const handleSignUp = async (req, res) => {
+  //get from body.
+  let filePath = req.file.path;
+  let name = req.body.name;
+  let pass = req.body.pass;
+  //initialize into new var.
+  let signUpInfo = {
+    user: name,
+    pass: pass,
+  };
+  //check if any of the fields are empty.
+  if (signUpInfo.user && signUpInfo.pass && req.file) {
```

Comment:
This and everything re: password hashing all seems like service logic to me, the controller should be very simple:
1. Do user and password exist in the request body?
2. Call service with user and password
3. Send response

Reply:
done